lib/manager/terragrunt/index.ts

```diff
@@ -0,0 +1,9 @@
+import * as hashicorpVersioning from '../../versioning/hashicorp';
+
+export { extractPackageFile } from './extract';
+
+export const defaultConfig = {
+  commitMessageTopic: 'Terragrunt dependency {{depNameShort}}',
+  fileMatch: ['\\.hcl$'],
```

Comment:
```suggestion
  fileMatch: ['(^|/)terragrunt\\.hcl$'],
```

I think we should be more strict here as terragrunt only reads `terragrunt.hcl` files by default, see [docs](https://terragrunt.gruntwork.io/docs/getting-started/configuration/#terragrunt-configuration-file)

A user can always override this if required.

Reply:
Actually @viceice I disagree because you can (and in the real world projects **you will**) have the possibility to [include other hcl](https://terragrunt.gruntwork.io/docs/getting-started/configuration/#configuration-parsing-order) files (which don't need to be called terragrunt.hcl) but those files still need to comply with the Terragrunt structure.

So you can have something like this:

- main.hcl

```hcl
terraform {
  source = "github.com/bgdanix/tfrepo//terraform/modules/moduleone?ref=0.0.9"
}

inputs = {
  name           = "mysql_${local.env}"
  instance_class = "db.t2.micro"

  allocated_storage = 20
  storage_type      = "standard"

  master_username = "admin"
  # TODO: To avoid storing your DB password in the code, set it as the environment variable TF_VAR_master_password
}
```

- terragrunt.hcl

```hcl
include {
  path = find_in_parent_folders("main.hcl")
}

inputs = {
  instance_class = "db.t3.2xlarge"
}
```

So that your main file has the "default" values and when you actually go and run terragrunt ti will indeed only look at the `terragrunt.hcl` file but it will include whatever is in the `include` part and just overwrite what has changed, which in this case would be just changing the DB instance size.

Reply:
What percentage of the world's HCL files would be terragrunt vs non-terragrunt?

Reply:
> Actually @viceice I disagree because you can (and in the real world projects **you will**) have the possibility

What are you disagreeing with? @viceice wrote "A user can always override this if required" which is true.

Reply:
What I'm disagreeing with is we should not tie it to terragrunt.hcl files only, but on .hcl files, based on the example I've just given. And also in my case the users can override it to include other extensions, but at least catch any .hcl files

Reply:
> 
> What percentage of the world's HCL files would be terragrunt vs non-terragrunt?

Hmm, indeed, this would make more sense now since HCL is developed by [HashiCorp](https://www.hashicorp.com) and it can be used in any of its products, not only Terragrunt. Ok, I'm fine with the proposed change by @viceice

Reply:
I'll tag you once I've pushed the changed as I'd like to also indicate this limitation in the documentation

Reply:
The question is not whether people do put terragunt configuration in alternate files, but about how many false hits we'll get if we default to `.hcl`. If terragrunt files represent only a small proportion of the world's `.hcl` files then Renovate users will end up doing a lot of wasted parsing. We could for example default the pattern `terragrunt.*\\.hcl$` which would mean any `.hcl` file which either has terragrunt in its filename or exists within any subfolder that includes `terragrunt` would be found

Reply:
Yes, you are right @rarkins , users will end up doing wasted parsing. Pushed the change now and also did two real tests by running the tool